dev/ProjectReunion_DLL/GetRawNotificationEventArgs.h

```diff
@@ -0,0 +1,34 @@
+﻿
+#include <pch.h>
+#include "PushNotificationReceivedEventArgs.h"
+#include "externs.h"
+
+using namespace winrt;
+using namespace winrt::Windows::Foundation;
+
+constexpr PCWSTR reunionPushServerArg = L"-ReunionPushServer";
+
+namespace GetRawNotificationEventArgs {
+    static Windows::Foundation::IInspectable GetRawNotificationEventArgs() {
+        Windows::Foundation::IInspectable args = nullptr;
+
+        std::wstring commandLineArgs = GetCommandLine();
+        auto found = commandLineArgs.find(reunionPushServerArg, 0);
+        if (found != std::wstring::npos)
+        {
+            {
+                auto lock = g_lock.lock();
+                THROW_HR_IF_NULL_MSG(E_UNEXPECTED, g_waitHandleForArgs, "PushNotificationManager::RegisterActivator has not been called.");
+            }
+
+            DWORD waitResult = WaitForSingleObject(g_waitHandleForArgs.get(), 1000);
+            if (waitResult == WAIT_OBJECT_0)
+            {
+                auto lock = g_lock.lock();
+                args = g_activatedEventArgs.as<Windows::Foundation::IInspectable>();
```

Comment:
Just std::swap should suffice. No need to cast either since args and g_activatedEvent args are of type IInspectable

Reply:
g_activatedEventArgs is PushNotificationReceivedEventArgs type. I can use swap but I would still need to make the explicit cast to IInspectable.

Reply:
So why do you need to cast at all? If you return a type of PushNotificationReceivedEventArgs, your method signature should automatically convert it to IInspectable. Right?

Reply:
So both args and g_activatedEventArgs can be of type PushNotificationReceivedEventArgs.

Reply:
I see now. The args variable that was returned was of type IInspectable instead of PushNotificationReceivedEventArgs. I changed args to type PushNotificationReceivedEventArgs and switched to std::swap.